unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/FeatureController/FeatureController.cs

```diff
@@ -32,27 +40,60 @@ public void Update()
         }
     }
 
+    public void LateUpdate()
+    {
+        foreach (Feature feature in activeFeatures)
+        {
+            feature.LateUpdate();
+        }
+    }
+
+    public void OnDestroy()
+    {
+        foreach (Feature feature in activeFeatures)
+        {
+            feature.Dispose();
+        }
+    }
+
     public void OnKernelConfigChanged(KernelConfigModel current, KernelConfigModel previous) { ApplyFeaturesConfig(current); }
 
     public void ApplyFeaturesConfig(KernelConfigModel config)
     {
-        HandleBuilderInWorld(config.features.enableBuilderInWorld);
+        HandleFeature<BIWMainController>(config.features.enableBuilderInWorld);
         currentConfig = config;
     }
 
-    private void HandleBuilderInWorld(bool isActive)
+    private void HandleFeature<T>(bool isActive) where T : Feature
     {
         if (isActive)
+            InitializeFeature<T>();
+        else
+            RemoveFeature<T>();
+    }
+
+    private void InitializeFeature<T>() where T : Feature
+    {
+        for (int i = 0; i < activeFeatures.Count; i++)
         {
-            if (builderInWorld != null)
+            if (activeFeatures[i].GetType() == typeof(T))
                 return;
-            builderInWorld = GameObject.Instantiate(builderInWorldFeaturePrefab);
-
         }
-        else
+
+        Feature feature = (Feature) Activator.CreateInstance(typeof (T));
+        feature.Initialize();
+        activeFeatures.Add(feature);
+    }
+
+    private void RemoveFeature<T>() where T : Feature
+    {
+        for (int i = 0; i < activeFeatures.Count; i++)
         {
-            if (builderInWorld != null)
-                GameObject.Destroy(builderInWorld);
+            if (activeFeatures[i].GetType() == typeof(T))
```

Comment:
`activeFeatures` already restricts the type to `Feature` so this check is redundant.

Reply:
If I am not mistaken ( I could be), this is getting the type of the Subclass type, (The subclass should inherit from Feature) so this way only the child is removed from the feature